jruby/neo4j/driver/ext/internal_session.rb

```diff
@@ -14,11 +14,15 @@ module InternalSession
         # work around jruby issue https://github.com/jruby/jruby/issues/5603
         Struct.new('Wrapper', :object)
 
-        %i[read write].each do |prefix|
-          define_method("#{prefix}_transaction") do |**config, &block|
-            check do
-              super(->(tx) { Struct::Wrapper.new(reverse_check { block.call(tx) }) }, to_java_config(Neo4j::Driver::TransactionConfig, **config)).object
-            end
+        # implementation of read_transaction, write_transaction, execute_read, execute_write
+        %i[read write].each do |mode|
+          define_method("execute_#{mode}") do |**config, &block|
+            execute_transaction(__method__, **config, &block)
+          end
+
+          define_method("#{mode}_transaction") do |**config, &block|
+            Neo4j::Driver::Internal::Deprecator.log_warning("#{mode}_transaction", "execute_#{mode}".to_sym, '6.0')
```

Comment:
You can use relative namespace
```suggestion
            Internal::Deprecator.log_warning("#{mode}_transaction", "execute_#{mode}".to_sym, '6.0')
```

Reply:
I tried `Internal::Deprecator.log_warning` but it raised a NameError, so left here `Driver::Internal::Deprecator.log_warning`